### PR TITLE
Reanalyse gender

### LINKED ISSUE #54: Include SampleID in files instead of BAMID
* BAM ID in file names is not needed

### LINKED ISSUE #61: Add used BED tracks (i.e. HC file) to VCF metadata
Specific BED tracks (i.e. HC file) used for CNV-calling is not in the VCF metadata

### LINKED ISSUE #64: Update README
File exons.hg19.full.tsv probably needs header with tab-separated:
chromosome	start	end	name

### LINKED ISSUE #66: Add female/male status option to submit_batch_exomedepth.py
It is not possible to perform re-analysis for a different male/female status at the moment.

### LINKED ISSUE #67: Add merge warning to CNV QC if merge sample
Add merge warning to CNV QC if merge sample
(if possible, also add sequencer type to warning)

### LINKED ISSUE #68: Make female/male choice in argparse ed_csv_to_vcf.py
Argument should be choice female/male only, nothing else.
Perhaps also improve if/else statement in code

### LINKED ISSUE #73: Alissa count file
Include Alissa count file as output in make_BEDdetail.py.

Alissa count file (tab sep):
chr start end DEL/DUP count

note: file should be 0-based

### LINKED ISSUE #75: make_BEDdetail.py gives incorrect count in header UCSC/IGV bed file
If sample did not pass QC, sample is not substracted from child/parent count

### PR DESCRIPTION
**Pull request by melferink**
•	Reanalyse male/female with warning in QC summar (fixed #66)
•	MERGE warning in QC summary for merged samples (fixed #67)
•	Included Alissa count file as output in make_BEDdetail.py (fixed #73)
•	Excluded QC failed samples in UCSC and IGV header. fixed #75
•	Male/female choice included ed_csv_to_vcf (fixed #68)
•	README update (fixed #64)
•	Added used git repo Dx_tracks to VCF metadata (fixed #61)
•	Changed bamID to sampleID in filenaming (fixed #54)
•	Split reference, gender, calling model in VCF metadata
•	Included full path to used reference genoom in VCF metadata

### CODE REVIEW

File: ExomeDepth/settings.py [hunk 1]
```diff
@@ -66,3 +68,13 @@
 correlation = 0.98
 number_calls = [35,200]
 del_dup_ratio = [15,85] 
+
+reanalysis_dic = {
+    "XXY":["female", "XXYMaleAsFemale"],
+    "X": ["male", "XFemaleAsMale"],
+    "asFemale": ["female", "AsFemale"],
+    "asMale": ["male", "AsMale"]
```
Comment (rernst) [inline ExomeDepth/settings.py:76]: Maybe remove 'as' and double gender description?

```
XXYFemale
XMale
Female
Male
```

File: ExomeDepth/settings.py.slangenspar [hunk 2]
```diff
@@ -24,7 +24,9 @@ singularity_mount_path = "/data:/data"
 
 # Location of target files
 refset_dir = "/data/databases/ExomeDepth_refset/"
-reffile_dir = "/data/software/production/Dx_tracks/ExomeDepth/"
+reffile_repo = "/data/software/production/Dx_tracks/"
+reffile_dir = "{}ExomeDepth/".format(reffile_repo)
```
Comment (rernst) [inline ExomeDepth/settings.py.slangenspar:28]: Add `/` after `{}`

File: ExomeDepth/settings.py.slangenspar [hunk 3]
```diff
@@ -66,3 +68,10 @@ igv_settings = {"ratio":[0, 1, 2], "log2ratio":[-2.5, 0, 2.5] }
 correlation = 0.98
 number_calls = [35,200]
 del_dup_ratio = [15,85] 
+
+reanalysis_dic = {
+    "XXY":["female", "XXYMaleAsFemale"],
+    "X0": ["male", "X0FemaleAsMale"],
+    "asFemale": ["female", "AsFemale"],
+    "asMale": ["male", "AsMale"]
+    }
```
Comment (rernst) [inline ExomeDepth/settings.py.slangenspar:77]: Different settings on slangespar? `X0`?

File: ExomeDepth/submit_batch_exomedepth.py [hunk 4]
```diff
@@ -115,16 +113,26 @@ def process(bam, args, refset_dic):
...
                     refset_dic[splitline[0]] = splitline[1]
-
+                if len(splitline)>2:  #Note: make sure reanalysis file must contain sampleid in first column and refset in second column. More elegant way to program here?
+                    if settings.reanalysis_dic[splitline[2]]: 
+                        if splitline[0] not in gender_dic: 
+                            gender_dic[splitline[0]] = settings.reanalysis_dic[splitline[2]][0]
+                        if splitline[0] not in  warning_dic:
+                            warning_dic[splitline[0]] = settings.reanalysis_dic[splitline[2]][1]
+                    else: 
+                        sys.exit("Warning {} in reanalysis file does not exict".format(splitline[2]))
```
Comment (rernst) [inline ExomeDepth/submit_batch_exomedepth.py:131]: Typo -> exict -> exist

Comment (rernst) [inline ExomeDepth/submit_batch_exomedepth.py:131]: I don't understand the sys.exit warning. You first check on line 124 whether reanalysis file contains at least 2 columns. If that is not the case you give a warning that the file does not exists. 

Secondly, create variables for splitline items. Very hard to follow what is precisely being done here.

**Review by rernst — CHANGES_REQUESTED** [review]
In general try to add a space after every `,` separating arguments.

For example:
'{}_{}'.format(derp, test)

File: ExomeDepth/README.md [hunk 5]
```diff
@@ -90,7 +90,7 @@ python submit_batch_exomedepth.py <input folder> <output folder> <samples(/threa
 
     Calculate coverage stats for each male/female folder for each population
     ``` bash
-    sh run_sambamba.sh <folder> <bed_file> <email>
+    sh run_sambamba.sh <path to sambamba executable> <folder> <bed_file> <email>
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/README.md:93]: Full to be sure

File: ExomeDepth/igv_xml_session.py [hunk 6]
```diff
@@ -32,12 +32,22 @@ def make_igvsession(igv_ed_umcu, igv_ed_hc, bam, vcf_hc, sample_id, vcf_SNV, axi
...
-    igv_ed_hc = "igv_tracks/HC_{0}_{1}_{2}_ref.igv".format(args.refdate, args.bam, args.runid)
-    vcf_hc = "HC/HC_{0}_{1}_{2}_exome_calls.vcf".format(args.refdate, args.bam, args.runid)
+  
+    igv_extension = "ref.igv"
+    if args.warning:
+        igv_extension = "{}_ref.igv".format(args.warning)
+
+    vcf_extension = "exome_calls.vcf"
+    if args.warning:
+        vcf_extension = "exome_calls_{}.vcf".format(args.warning)
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/igv_xml_session.py:46]: VCF file is not known (could be parsed if needed), but current (seperate) argument are needed for IGV, BAM, and SNV VCF anyways.

igv_extension ánd vcf_extension placed within the same args.warning is possible

File: ExomeDepth/make_BEDdetail.py [hunk 7]
```diff
@@ -19,7 +19,8 @@ def make_merge_dic(merge_samples):
     return merge_dic
 
 def slice_vcf(args, merge_dic):
-    vcf_files = subprocess.getoutput("find {input} -type f -iname \"*vcf\"".format(input=args.inputfolder)).split()
+    vcf_files = subprocess.getoutput("find -L {input} -type f -iname \"*vcf\"".format(input=args.inputfolder)).split()
+    print(vcf_files)
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/make_BEDdetail.py:23]: forgot to remove...

File: ExomeDepth/settings.py (hunk 1 above)
Comment (melferink) [reply ExomeDepth/settings.py:76]: Nope, this is all I could bargain for ;)

File: ExomeDepth/settings.py.slangenspar (hunk 3 above)
Comment (melferink) [reply ExomeDepth/settings.py.slangenspar:77]: Should have been removed, thanks. 0 is not a dummy proof character ;)

File: ExomeDepth/igv_xml_session.py [hunk 8]
```diff
@@ -13,11 +13,11 @@ def make_igvsession(igv_ed_umcu, igv_ed_hc, bam, vcf_hc, sample_id, vcf_SNV, axi
...
-    substitute_dic={'session_var' : new_session, 'igv_ed_umcu' : igv_ed_umcu, 'igv_ed_hc' : igv_ed_hc, 
-                    'bam':bam, 'vcf_hc' : vcf_hc, 'sample_id' : sample_id, 'igv_ed_hc_test' : igv_ed_hc_test,
-                    'igv_ed_umcu_test' : igv_ed_umcu_test, 'bam_coverage' : bam_coverage, 'bam_junctions' : bam_junctions, 
-                    'vcf_SNV' : vcf_SNV, 'min_axis' : min_axis, 'mid_axis' : mid_axis, 'max_axis' : max_axis,
-                    'ratioid_UMCU' : ratioid_UMCU, 'ratioid_HC': ratioid_HC 
+    substitute_dic={'session_var':new_session, 'igv_ed_umcu':igv_ed_umcu, 'igv_ed_hc':igv_ed_hc, 
+                    'bam':bam, 'vcf_hc':vcf_hc, 'sample_id':sample_id, 'igv_ed_hc_test':igv_ed_hc_test, 
+                    'igv_ed_umcu_test':igv_ed_umcu_test, 'bam_coverage':bam_coverage, 'bam_junctions':bam_junctions,  
+                    'vcf_SNV':vcf_SNV, 'min_axis':min_axis, 'mid_axis':mid_axis, 'max_axis':max_axis, 
+                    'ratioid_UMCU':ratioid_UMCU, 'ratioid_HC':ratioid_HC
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:20]: Space missing after `:` and around `=`.

```python
 substitute_dic = {'session_var': new_session}
```

File: ExomeDepth/make_BEDdetail.py [hunk 9]
```diff
@@ -18,9 +18,16 @@ def make_merge_dic(merge_samples):
 
     return merge_dic
 
+def is_child(sampleid):
+    if any([child_tag in sampleid for child_tag in ['CM', 'CF', 'CO']]):
+        return True
+
+def is_parent(sampleid):
+    if any([parent_tag in sampleid for parent_tag in ['PM', 'PF', 'PO']]):
+        return True
```
Comment (rernst) [inline ExomeDepth/make_BEDdetail.py:27]: Do you need both? If a sample is not a child -> it is a parent?

File: ExomeDepth/run_ExomeDepth.py [hunk 10]
```diff
@@ -139,48 +150,52 @@ def multiprocess_call(multiprocess_list):
...
+        multiprocess_list[6], 
+        multiprocess_list[0], 
+        args.refset, 
+        args.sample, 
+        args.run
         )
-    if args.warning:
-        action = "{action} --warning {warning}".format(action=action, warning=args.warning)
+    action = ("python {csv2vcf} {inputcsv} {refset} {model} {gender} "
+        " {sampleid} {template} {runid} ").format(
```
Comment (rernst) [inline ExomeDepth/run_ExomeDepth.py:187]: Maybe add new line after `(`?

```python
action = (
    "python {csv2vcf} {inputcsv} {refset} {model} {gender} "
    " {sampleid} {template} {runid} ").format()
)
```

**Review by rernst — CHANGES_REQUESTED** [review]
Some small things left.

File: ExomeDepth/make_BEDdetail.py (hunk 9 above)
Comment (melferink) [reply ExomeDepth/make_BEDdetail.py:27]: Could be an option, but them the otherway around. If not parent than child: Parent need to be as 'clean' as possible as these are 'unaffected'. 
But to be sure, I think we should keep both for now, and exclude samples that have unknown status (i.e. typos / research/ validation samples).
We could change this if we can query status from a db.
